engine/table/src/main/java/io/deephaven/engine/table/impl/TimeTable.java

```diff
@@ -207,6 +208,9 @@ public DateTime get(long rowKey) {
         }
 
         private long computeNanos(long rowKey) {
+            if (rowKey < 0) {
+                return QueryConstants.NULL_LONG;
+            }
```

Comment:
This is probably the wrong fix:
1. `computeNanos` is used from `fillChunk`, which cannot have invalid row keys, and so you shouldn't burden it with the `rowKey` check
2. You have not addressed `get`, just `getLong`.

I would revert this change, and instead fix `io.deephaven.engine.table.impl.TimeTable.SyntheticDateTimeSource#get`, `io.deephaven.engine.table.impl.TimeTable.SyntheticDateTimeSource#getLong`, `io.deephaven.engine.table.impl.TimeTable.SyntheticDateTimeSource.SyntheticDateTimeAsLongSource#get`, and `io.deephaven.engine.table.impl.TimeTable.SyntheticDateTimeSource.SyntheticDateTimeAsLongSource#getLong`.